pint/quantity.py

```diff
@@ -79,6 +78,262 @@ def wrapped(self, *args, **kwargs):
+    return np.searchsorted(a, v, *args, **kwargs)
+
+# Handle single unit argument operations (axis ops, aggregations, etc.)
+for func_str in ['expand_dims', 'squeeze', 'rollaxis', 'moveaxis', 'fix', 'around', 'diagonal',
+                 'mean', 'ptp', 'ravel', 'round_', 'sort', 'median', 'nanmedian', 'transpose',
+                 'flip', 'copy', 'trim_zeros', 'average', 'nanmean']:
+    func = getattr(np, func_str)
+    @implements(func)
+    def implementation(a, *args, **kwargs):
+        (a,), output_wrap = unwrap_and_wrap_consistent_units(a)
+        return output_wrap(func(a, *args, **kwargs))
+
+# Handle atleast_nd functions
+for func_str in ['atleast_1d', 'atleast_2d', 'atleast_3d']:
+    func = getattr(np, func_str)
+    @implements(func)
+    def implementation(*arrays):
+        stripped_arrays, _ = convert_to_consistent_units(None, *arrays)
+        arrays_magnitude = func(*stripped_arrays)
+        if len(arrays) > 1:
+            return [array_magnitude if not isinstance(original, Quantity)
+                    else original._REGISTRY.Quantity(array_magnitude, original.units)
+                    for array_magnitude, original in zip(arrays_magnitude, arrays)]
+        else:
+            output_unit = arrays[0].units
+            return output_unit._REGISTRY.Quantity(arrays_magnitude, output_unit)
+
+# Handle single-argument consistent unit functions
+for func_str in ['block', 'hstack', 'vstack', 'dstack', 'column_stack']:
+    implement_func(func_str, input_units='all_consistent', output_unit='match_input')
+
+implement_func('unwrap', input_units='rad', output_unit='match_input')
+
+for func_str in ['cumprod', 'cumproduct', 'nancumprod']:
+    implement_func(func_str, input_units='dimensionless', output_unit='match_input')
+
+for func_str in ['size', 'isreal', 'iscomplex', 'shape', 'ones_like', 'zeros_like',
+                 'empty_like', 'argsort', 'argmin', 'argmax', 'alen', 'ndim', 'nanargmax',
+                 'nanargmin', 'count_nonzero', 'nonzero', 'result_type']:
+    implement_func(func_str, input_units=None, output_unit=None)
+
+# TODO: Verify all these below with non-united other arguments \/ !!
+
+for func_str in ['std', 'nanstd', 'sum', 'nansum', 'cumsum', 'nancumsum']:
+    implement_func(func_str, input_units=None, output_unit='sum')
+    
+for func_str in ['cross', 'trapz', 'dot']:
+    implement_func(func_str, input_units=None, output_unit='prod')
+
+for func_str in ['diff', 'ediff1d',]:
+    implement_func(func_str, input_units=None, output_unit='delta')
+
+for func_str in ['gradient', ]:
+    implement_func(func_str, input_units=None, output_unit='delta,div')
+
+for func_str in ['var', 'nanvar']:
+    implement_func(func_str, input_units=None, output_unit='variance')
+
+# TODO: broadcast_to (how to handle subok?)
+# TODO: 'amax', 'amin', 'nanmax', 'nanmin' (version dependent signatures)
+# TODO: nan_to_num (expected behavior with input values, sensible defaults?)
+
```

Comment:
Since it is being used a few times in xarray's codebase: can we also add a TODO for `result_type`?